LISA/Executor.cpp

```diff
@@ -233,52 +245,87 @@ void Executor::doInstall(std::string type,
+    setProgress(0, OperationStage::UNTARING);
+    INFO("unpacking ", tmpFilePath, "to ", appsPath);
     Archive::unpack(tmpFilePath, appsPath);
 
     auto appStoragePath = Filesystem::getAppsStorageDir() + appSubPath;
-    TRACE_L1("creating storage %s ", appStoragePath.c_str());
+    INFO("creating storage ", appStoragePath);
     Filesystem::ScopedDir scopedAppStorageDir{appStoragePath};
 
     // everything went fine, mark app directories to not be removed
     scopedAppDir.commit();
     scopedAppStorageDir.commit();
 
+    setProgress(0, OperationStage::UPDATING_DATABASE);
+
     // TODO add entry to database
 
+    setProgress(0, OperationStage::FINISHED);
+
     // TODO invoke maintenace and cleanup
     // TODO notify status
 
-    TRACE_L1("finished, notify status");
+    INFO("finished");
 }
 
 void Executor::doUninstall(std::string type, std::string id, std::string version, std::string uninstallType)
 {
-    TRACE_L1("type=%s idName=%s version=%s uninstallType=%s", type.c_str(), id.c_str(), version.c_str(),
-        uninstallType.c_str());
+    INFO("type=", type, " id=", id, " version=", version, " uninstallType=", uninstallType);
 
     // TODO remove from database
 
     // TODO should "id" be also removed? what if two versions are installed?
     auto appSubPath = Filesystem::createAppPath(type, id, version);
 
     auto appPath = Filesystem::getAppsDir() + appSubPath;
-    TRACE_L1("removing %s ", appPath.c_str());
+    INFO("removing ", appPath);
     Filesystem::removeDirectory(appPath);
 
     if (uninstallType == "full") {
         auto appStoragePath = Filesystem::getAppsStorageDir() + appSubPath;
-        TRACE_L1("removing storage directory %s ", appStoragePath.c_str());
+        INFO("removing storage directory ", appStoragePath);
         Filesystem::removeDirectory(appStoragePath);
     }
 
     // TODO invoke maintenace and cleanup
     // TODO notify status
 
-    TRACE_L1("finished, notify status");
+    INFO("finished");
+}
+
+void Executor::setProgress(int stagePercent, OperationStage stage)
+{
+    constexpr auto STAGES = enumToInt(OperationStage::COUNT);
+    const std::array<int, STAGES> stageBase = {{0, 90, 95, 100}};
+    const std::array<double, STAGES> stageFactor = {{90.0/100, 5.0/100, 5.0/100, 0}};
+
+    int stageIndex = enumToInt(stage);
+    int resultPercent = stageBase[stageIndex] + (static_cast<int>(stagePercent * stageFactor[stageIndex]));
+
+    INFO("overall: ", resultPercent, "% from stage: ", stage, " progress: ", stagePercent, "%", " ");
+
+    LockGuard lock{taskMutex};
+    currentTask.progress = resultPercent;
+}
+
+std::ostream& operator<<(std::ostream& out, const Executor::Task& task)
+{
+    return out << "task[" << task.handle << "]";
+}
+
+std::ostream& operator<<(std::ostream& out, Executor::OperationStage stage)
+{
+    const std::array<const char*, enumToInt(Executor::OperationStage::COUNT)> stages = {{
+            "DOWNLOADING",
+            "UNTARING",
```

Comment:
Maybe EXTRACT? Or it will be always tar archive?

Reply:
Hard to say how it will change in the future ;) for now docs explicitly mention untar, so let's leave it as it is.